tests/test_feature_compare.py

```diff
@@ -13,17 +16,23 @@ def test_compare(meta_df):
     obs = compare(meta_df, clone, right_label='meta_df', left_label='clone')
 
     exp = pd.DataFrame([
-        ['Primary Energy', 'EJ/y', 2005, 2, 1],
-        ['Primary Energy|Coal', 'EJ/y', 2005, np.nan, 0.5],
-        ['Primary Energy|Coal', 'EJ/y', 2010, np.nan, 3],
-        ['Primary Energy|Gas', 'EJ/y', 2005, 0.5, np.nan],
-        ['Primary Energy|Gas', 'EJ/y', 2010, 3, np.nan],
+        ['Primary Energy', 'EJ/y', dt.datetime(2005, 6, 17), 2, 1],
+        ['Primary Energy|Coal', 'EJ/y', dt.datetime(2005, 6, 17), np.nan, 0.5],
+        ['Primary Energy|Coal', 'EJ/y', dt.datetime(2010, 7, 21), np.nan, 3],
+        ['Primary Energy|Gas', 'EJ/y', dt.datetime(2005, 6, 17), 0.5, np.nan],
+        ['Primary Energy|Gas', 'EJ/y', dt.datetime(2010, 7, 21), 3, np.nan],
     ],
-        columns=['variable', 'unit', 'year', 'meta_df', 'clone'],
+        columns=['variable', 'unit', 'time', 'meta_df', 'clone'],
     )
     exp['model'] = 'model_a'
     exp['scenario'] = 'scen_a'
     exp['region'] = 'World'
-    exp = exp.set_index(IAMC_IDX + ['year'])
+    time_col = 'time'
+    if "year" in meta_df.data.columns:
+        exp["year"] = exp["time"].apply(lambda x: x.year)
```

Comment:
use the new function `swap_time_for_year()`?

Reply:
can't do that here as `exp` has nan's so can't be cast to `IamDataFrame`

Reply:
got it, thanks!